test/unit/org/apache/cassandra/index/sai/cql/AllTypesSimpleEqTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import accord.utils.Gen;
+import accord.utils.Gens;
+import org.apache.cassandra.cql3.CQL3Type;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.AsciiType;
+import org.apache.cassandra.db.marshal.DecimalType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.utils.AbstractTypeGenerators;
+import org.apache.cassandra.utils.Generators;
+import org.quicktheories.generators.SourceDSL;
+
+@RunWith(Parameterized.class)
+public class AllTypesSimpleEqTest extends AbstractSimpleEqTest
+{
+    private static final Map<AbstractType<?>, Long> LARGE_DOMAIN_FAILING_SEEDS =
+        Map.of(AsciiType.instance, -5097559362029524309L, UTF8Type.instance, -4379508235061872764L);
+    private static final Map<AbstractType<?>, Long> SHORT_DOMAIN_FAILING_SEEDS =
+        Map.of(AsciiType.instance, -5097559362029524309L, UTF8Type.instance, -4379508235061872764L);
+    private final AbstractType<?> type;
+
+    public AllTypesSimpleEqTest(AbstractType<?> type)
+    {
+        this.type = type;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data()
+    {
+        return StorageAttachedIndex.SUPPORTED_TYPES.stream()
+                                                   .map(CQL3Type::getType)
+                                                   // TODO: Track down unicode character edge cases...
+                                                   .filter(t -> t != UTF8Type.instance)
```

Comment:
don't you need to filter ascii as well?

Reply:
Ascii should pass without issue, since it doesn't have the problematic unicode characters.